Source/AwsCommonRuntimeKit/sdkutils/endpoint/EndpointsRequestContext.swift

```diff
@@ -48,6 +48,34 @@ public class EndpointsRequestContext {
         }
     }
 
+    private func withByteCursorArrayFromStringArray<R>(
+        _ arg: [String], _ body: (UnsafeMutablePointer<aws_byte_cursor>, Int) -> R) -> R {
+        withArrayOfCStrings(arg) { cStrArr in
+            var cursors = cStrArr.map { aws_byte_cursor_from_c_str($0) }
+            let len = cursors.count
+            return cursors.withUnsafeMutableBufferPointer { cursorsPtr in 
+                return body(cursorsPtr.baseAddress!, len)
+            }
+        }
+    }
+
+    /// Add a string array endpoint parameter to the request context
+    /// - Parameters:
+    ///   - name: The name of the parameter
+    ///   - value: The value of the parameter
+    public func add(name: String, value: [String]?) throws {
+        guard let value = value else {
+            return
+        }
```

Comment:
We should guard against empty array or have a test to make sure that it works as expected.

Reply:
let me add a test. empty array is a valid input